bitnami/etcd/templates/cronjob-defrag.yaml

```diff
@@ -0,0 +1,92 @@
+  annotations: {{- include "common.tplvalues.render" ( dict "value" $annotations "context" $) | nindent 4 }}
+  {{- end }}
+spec:
+  {{- if .Values.defrag.cronjob.startingDeadlineSeconds }}
+  startingDeadlineSeconds: {{ .Values.defrag.cronjob.startingDeadlineSeconds }}
+  {{- end }}
+  concurrencyPolicy: {{ .Values.defrag.cronjob.concurrencyPolicy }}
+  schedule: {{ .Values.defrag.cronjob.schedule | quote }}
+  suspend: {{ .Values.defrag.cronjob.suspend }}
+  successfulJobsHistoryLimit: {{ .Values.defrag.cronjob.successfulJobsHistoryLimit }}
+  failedJobsHistoryLimit: {{ .Values.defrag.cronjob.failedJobsHistoryLimit }}
+  jobTemplate:
+    spec:
+      template:
+        metadata:
+          {{- $mergedLabels := mergeOverwrite (dict) .Values.commonLabels .Values.defrag.cronjob.podLabels -}}
+          labels: {{- include "common.labels.standard" ( dict "customLabels" $mergedLabels "context" $ ) | nindent 10 }}
+          {{- if .Values.defrag.cronjob.podAnnotations }}
+          annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.defrag.cronjob.podAnnotations "context" $) | nindent 12 }}
+          {{- end }}
+        spec:
+          {{- if .Values.defrag.cronjob.activeDeadlineSeconds }}
+          activeDeadlineSeconds: {{ .Values.defrag.cronjob.activeDeadlineSeconds }}
+          {{- end }}
+          restartPolicy: {{ .Values.defrag.cronjob.restartPolicy }}
+          {{- if .Values.podSecurityContext.enabled }}
+          securityContext: {{- include "common.compatibility.renderSecurityContext" (dict "secContext" .Values.podSecurityContext "context" $) | nindent 12 }}
+          {{- end }}
+          containers:
+            - name: etcd-defrag
+              image: {{ include "etcd.image" . }}
+              imagePullPolicy: "IfNotPresent"
+              {{- if .Values.containerSecurityContext.enabled }}
+              securityContext: {{- include "common.compatibility.renderSecurityContext" (dict "secContext" .Values.containerSecurityContext "context" $) | nindent 16 }}
+              {{- end }}
+              env:
+                - name: BITNAMI_DEBUG
+                  value: {{ ternary "true" "false" (or .Values.image.debug .Values.diagnosticMode.enabled) | quote }}
+                - name: ETCDCTL_API
+                  value: "3"
+                - name: ETCD_ON_K8S
+                  value: "yes"
+                - name: MY_STS_NAME
+                  value: {{ include "common.names.fullname" . | quote }}
+                {{- $releaseNamespace := .Release.Namespace }}
+                {{- $etcdFullname := include "common.names.fullname" . }}
+                {{- $etcdHeadlessServiceName := (printf "%s-%s" $etcdFullname "headless" | trunc 63 | trimSuffix "-") }}
+                {{- $clusterDomain := .Values.clusterDomain }}
+                - name: ETCD_CLUSTER_DOMAIN
+                  value: {{ printf "%s.%s.svc.%s" $etcdHeadlessServiceName $releaseNamespace $clusterDomain | quote }}
```

Comment:
Are these env. vars required?

Reply:
@juan131 

> Are these env. vars required?

Actually I copy-pasted them from another etcd CronJob template ([disasterRecovery](https://github.com/bitnami/charts/blob/8435f88ae01cc20203022eb97904ee3abeb96467/bitnami/etcd/templates/cronjob.yaml#L83-L96)), but I'm not sure if they're all required 🤔 

From my understanding of [etcd env vars](https://hub.docker.com/r/bitnami/etcd):
- `ETCDCTL_API` can be removed favoring the default value (`3`);
- `ETCD_ON_K8S` can be removed;
- `MY_STS_NAME` I don't know if this does something ([#54769](https://github.com/bitnami/containers/issues/54769));
- `ETCD_CLUSTER_DOMAIN` this is used for DNS-based peer discovery. Since the `--endpoint` flag is already specified, _maybe_ this can be removed;

I currently have no way to test these other than on local-minikube environment, which I'll try soon.
If it turns out they're not needed, as a general solution we could:
- just remove them;
- remove them but have an extra `defrag.cronjob.env` field that allows for a custom configuration

Let me know if you think that makes sense 🙂

Reply:
I've tested the chart changes on our GKE cluster (`bitnami/apisix` + `bitnami/etcd` with 3 replicas) and removing these 4 env vars (`ETCDCTL_API`, `ETCD_ON_K8S`, `MY_STS_NAME`, `ETCD_CLUSTER_DOMAIN`) doesn't seem to impact the defrag CronJob

Reply:
awesome! thanks